lib/class-wp-duotone-gutenberg.php

```diff
@@ -0,0 +1,375 @@
+<?php
+/**
+ * WP_Duotone_Gutenberg class
+ *
+ * @package gutenberg
+ * @since 6.3.0
+ */
+
+/**
+ * Manages which duotone filters need to be output on the page.
+ *
+ * @access public
+ */
+class WP_Duotone_Gutenberg {
+	/**
+	 * An array of Duotone presets from global, theme, and custom styles.
+	 *
+	 * Example:
+	 * [
+	 *      'blue-orange' =>
+	 *          [
+	 *              'slug'  => 'blue-orange',
+	 *              'colors' => [ '#0000ff', '#ffcc00' ],
+	 *          ]
+	 *      ],
+	 *      …
+	 * ]
+	 *
+	 * @since 6.3.0
+	 * @var array
+	 */
+	protected static $global_styles_presets = array();
+
+	/**
+	 * An array of block names from global, theme, and custom styles that have duotone presets. We'll use this to quickly
+	 * check if a block being rendered needs to have duotone applied, and which duotone preset to use.
+	 *
+	 * Example:
+	 *  [
+	 *      'core/featured-image' => 'blue-orange',
+	 *       …
+	 *  ]
+	 *
+	 * @since 6.3.0
+	 * @var array
+	 */
+	protected static $global_styles_block_names = array();
+
+	/**
+	 * An array of Duotone SVG and CSS ouput needed for the frontend duotone rendering based on what is
+	 * being ouptput on the page. Organized by a slug of the preset/color group and the information needed
```

Comment:
```suggestion
	 * An array of Duotone SVG and CSS output needed for the frontend duotone rendering based on what is
	 * being output on the page. Organized by a slug of the preset/color group and the information needed
```

Couple of small typos.